### ISSUE TITLE
segmentation fault

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#275]
**Reported by shirish on 28 Jun 37460709 09:46 UTC**
Boinc Client invokes a segmentation fault when detaching from project 
```
./boinc --detach_project http://www.worldcommunitygrid.org/
2007-06-17 17:45:22 [Starting BOINC client version 5.10.2 for i686-pc-linux-gnu
2007-06-17 17:45:22 [---](---]) log flags: task, file_xfer, sched_ops
2007-06-17 17:45:22 [Libraries: libcurl/7.16.0 OpenSSL/0.9.8d zlib/1.2.3
2007-06-17 17:45:22 [---](---]) Data directory: /home/shirish/boinc/BOINC
2007-06-17 17:45:22 [Processor: 1 GenuineIntel Intel(R) Pentium(R) 4 CPU 1.80GHz [Family 15 Model 1 Stepping 2](---])
2007-06-17 17:45:22 [Processor features: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm up
2007-06-17 17:45:22 [---](---]) Memory: 622.44 MB physical, 1.92 GB virtual
2007-06-17 17:45:22 [Disk: 62.33 GB total, 53.39 GB free
2007-06-17 17:45:22 [http://www.worldcommunitygrid.org/](---]) Resetting project
2007-06-17 17:45:22 [http://www.worldcommunitygrid.org/] Detaching from project
SIGSEGV: segmentation violation
*** glibc detected *** ./boinc: malloc(): memory corruption: 0x0823e608 ***
```

**Comment by romw** [comment]
**Commented by shirish on 1 Apr 37461087 22:40 UTC**
Please format it properly, I tried to format it after Kathryn set stuff like owner & component and Ageless changed the description. 

Cheers !

**Comment by romw** [comment]
**Commented by Nicolas on 22 Jul 37462427 23:33 UTC**
Log with more debugging flags:

```
 ./boinc --detach_project http://isaac.ssl.berkeley.edu/alpha
2007-06-18 07:54:50 [Unparsed tag in cc_config.xml: <work_request_factor>
UNRECOGNIZED: work_request_factor
2007-06-18 07:54:50 [---](---]) Starting BOINC client version 5.10.7 for i686-pc-linux-gnu
2007-06-18 07:54:50 [log flags: task, file_xfer, sched_ops, cpu_sched, cpu_sched_debug, task_debug
2007-06-18 07:54:50 [---](---]) log flags: work_fetch_debug, unparsed_xml, state_debug, file_xfer_debug, sched_op_debug
2007-06-18 07:54:50 [log flags: http_debug, time_debug, http_xfer_debug, app_msg_send, network_status_debug
2007-06-18 07:54:50 [---](---]) log flags: checkpoint_debug
2007-06-18 07:54:50 [Libraries: libcurl/7.16.0 OpenSSL/0.9.8d zlib/1.2.3
2007-06-18 07:54:50 [---](---]) Data directory: /home/shirish/boinc/BOINC
2007-06-18 07:54:50 [[state_debug](---]) set dirty: Set mode
2007-06-18 07:54:50 [[state_debug](---]) set dirty: Set mode
2007-06-18 07:54:50 [Processor: 1 GenuineIntel Intel(R) Pentium(R) 4 CPU 1.80GHz [Family 15 Model 1 Stepping 2](---])
2007-06-18 07:54:50 [Processor features: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm up
2007-06-18 07:54:50 [---](---]) Memory: 622.44 MB physical, 1.92 GB virtual
2007-06-18 07:54:50 [Disk: 62.33 GB total, 52.36 GB free
2007-06-18 07:54:50 [---](---]) [CLIENT_STATE::write_state_file(): Writing state file
2007-06-18 07:54:50 [---](status_debug]) [CLIENT_STATE::write_state_file(): Done writing state file
2007-06-18 07:54:50 [---](status_debug]) [CLIENT_STATE::print_summary(): Client state summary:
2007-06-18 07:54:50 [---](state_debug]) 1 projects:
2007-06-18 07:54:50 [    http://isaac.ssl.berkeley.edu/alpha/ min RPC 2043.482491.0 seconds from now
2007-06-18 07:54:50 [---](---]) 2 file_infos:
2007-06-18 07:54:50 [    alpha_40_0.png status:1 inactive
2007-06-18 07:54:50 [---](---])     alpha_290_3.png status:1 inactive
2007-06-18 07:54:50 [0 app_versions
2007-06-18 07:54:50 [---](---]) 0 workunits
2007-06-18 07:54:50 [0 results
2007-06-18 07:54:50 [---](---]) 0 persistent file xfers
2007-06-18 07:54:50 [0 active tasks
2007-06-18 07:54:50 [BOINC alpha test](---]) Resetting project
2007-06-18 07:54:50 [[cpu_sched_debug](---]) Request CPU reschedule: exit_tasks
2007-06-18 07:54:50 [[state_debug](---]) CLIENT_STATE::garbage_collect(): deleting file alpha_40_0.png
2007-06-18 07:54:50 [[state_debug](---]) CLIENT_STATE::garbage_collect(): deleting file alpha_290_3.png
2007-06-18 07:54:50 [[state_debug](---]) CLIENT_STATE::print_summary(): Client state summary:
2007-06-18 07:54:50 [1 projects:
2007-06-18 07:54:50 [---](---])     http://isaac.ssl.berkeley.edu/alpha/ min RPC 2043.482491.0 seconds from now
2007-06-18 07:54:50 [0 file_infos:
2007-06-18 07:54:50 [---](---]) 0 app_versions
2007-06-18 07:54:50 [0 workunits
2007-06-18 07:54:50 [---](---]) 0 results
2007-06-18 07:54:50 [0 persistent file xfers
2007-06-18 07:54:50 [---](---]) 0 active tasks
2007-06-18 07:54:50 [[status_debug](---]) CLIENT_STATE::write_state_file(): Writing state file
2007-06-18 07:54:50 [[status_debug](---]) CLIENT_STATE::write_state_file(): Done writing state file
2007-06-18 07:54:50 [alpha test](BOINC) Detaching from project
2007-06-18 07:54:50 [[status_debug](---]) CLIENT_STATE::write_state_file(): Writing state file
2007-06-18 07:54:50 [[status_debug](---]) CLIENT_STATE::write_state_file(): Done writing state file
SIGSEGV: segmentation violation
*** glibc detected *** ./boinc: malloc(): memory corruption: 0x0823e180 ***
```

**Comment by romw** [comment]
**Commented by davea on 13 Aug 37513929 17:20 UTC**
can't reproduce, 6 July 07

**Comment by romw** [comment]
**Commented by davea on 18 May 37580135 03:06 UTC**
(In [13243]) - client: don't crash if use --detach_project option
    (fixes #283)
- client: canonicalize URLs in all cmdlines
- db_update.php: don't prompt for DB name/host; get it from config file
    (fixes #356)

**Comment by romw** [comment]
**Commented by romw on 11 Dec 37604665 14:40 UTC**
(In [David  31 July 2007
    - client: don't crash if use --detach_project option
        (fixes #283)
    - client: canonicalize URLs in all cmdlines
    - db_update.php: don't prompt for DB name/host; get it from config file
        (fixes #356)

    html/ops/
        db_update.php
    client/
        cs_cmdline.C

Charlie 31 July 2007
    - Graphics API: Don't render graphics when application is suspended 
        to reduce CPU usage.
    
    api/
        x_opengl.C

David  1 Aug 2007
    - GUI RPC: added get_global_prefs_network()
        and get_global_prefs_working()

    lib/
        gui_rpc_client.h
        gui_rpc_client_ops.C
    client/
        gui_rpc_server_ops.C
        cs_prefs.C

Charlie 1 Aug 2007
    - MGR: Populate Simple and Advanced Preference dialogs with current 
        values of preferences (including any overrides), not with 
        defaults or old (possibly stale) values.
        
    lib/
        gui_rpc_client_ops.C
    clientgui/
        DlgAdvPreferences.cpp
        sg_DlgPreferences.cpp

Charlie 2 Aug 2007
    - GUI RPC: rename get_global_prefs_network() to get_global_prefs_file() 
        to make it clear this returns the contents of the file which was 
        obtained from the network (i.e., web-based preferences) and not just 
        the prefernces affecting network connectivity.
    - GUI RPC: Add get_global_prefs_working_struct().
    - MGR: call get_global_prefs_working_struct() for both preference dialogs.
    - MGR: Dialup Manager updates current preferences from client before using 
        them.

     client/
        gui_rpc_server_ops.C
    clientgui/
        BOINCDialupManager.cpp
        DlgAdvPreferences.cpp
        sg_DlgPreferences.cpp
    lib/
        gui_rpc_client.h
        gui_rpc_client_ops.C

Rom    2 Aug 2007
    - DIAG: Disable symbol server technology for applications where a symbol
        server isn't specified.
    - MGR: Provide some additional text to the begining of the attach to project
        wizard when the client is attached to an account manager. Warn the user
        that the AMS will not know about the project or its stats.
    - MGR: Fix dialog titles for the Simple GUI dialogs.
    - MGR: Eliminate bogus error messages when using the default skin.
    - MGR: Introduce new logo into the manager
    - MGR: Introduce the notion of the syncronize button in the Simple GUI.
        NOTE: This button will replace the 'attach to project' button when
        the manager detects that the client is attached to an account manager.
    
    client/win/res/
        icon.ico
    clientgui/
        DlgAbout.cpp
        sg_DlgMessages.cpp
        sg_DlgPreferences.cpp
        SkinManager.cpp, .h
        WelcomePage.cpp
    clientgui/res/
        boinc.xpm
        boinc_logo.xpm
        BOINCGUIApp.ico
    clientgui/res/templates/
        boinc.ico
        boinc16.bmp
        boinc32.bmp
        boinc48.bmp
        boinc128.bmp
        boinc256.bmp
        boinc_logo.gif
    lib/
        stackwalker_win.cpp

Eric K 	 06 Aug 2007
    -added new functions boinc_suspend_graphics_thread() and
     boinc_resume_graphics_thread(), so apps can suspend the graphics
     threads for critical sections without suspending all threads.
     Only implemented under WIN32 thus far.
    api/
        graphics_lib.[Ch](13273]))

Rom    7 Aug 2007
    - MGR: Respect the disallow_attach and simple_gui_only status
        flags in both the simple GUI and advanced GUI.
    - MGR: Enable the syncronize button in the simple GUI which
        just causes an account manager update request.
    - MGR: Fix an overactive assert in debug builds

    clientgui/
        AdvancedFrame.cpp
        Events.h
        sg_BoincSimpleGUI.cpp
        sg_ProjectsComponent.cpp, .h
        SkinManager.cpp
        WelcomePage.cpp
    clientgui/res/skins/default/graphic
        syncronize_button.xpm (Added)
        syncronize_clicked_button.xpm (Added)
    clientgui/res/templates/skins
        btnSynronize.png (Added)
        btnSynronizeClick.png (Added)

Rom    8 Aug 2007
    - MGR: Update taskbar icons.
    
    clientgui/res/
        boinc.xpm
        boincdisconnect.xpm
        boincsnooze.xpm
    clientgui/res/templates/
        boincsnooze.ico
        boincdisconnect.ico
